mysticeti-core/src/consensus/base_committer.rs

```diff
@@ -145,16 +147,24 @@ impl BaseCommitter {
         &self,
         potential_certificate: &Data<StatementBlock>,
         leader_block: &Data<StatementBlock>,
+        all_votes: &mut HashMap<BlockReference, bool>,
```

Comment:
Lets add comment/warning here that `all_votes` must be only preserved when considered the same `leader_block` - this is the case now in the code, but I just want to make sure in the future people understand that this cache can not be made into global variable for example, or reused across calls regarding different blocks